lib/entities/version-file.js

```diff
@@ -0,0 +1,37 @@
+import moment from 'moment';
+
+import { SemanticFile } from './semantic-file';
+
+const TIMESTAMP_FORMAT = 'YYYYMMDDhhmmss';
+const TIMESTAMP_REGEX = /(\d{14})(?!.*\d{14})/;
+
+/**
+ *  Class representation of a versioned file.
+ */
+export class VersionFile extends SemanticFile {
```

Comment:
Maybe `VersionedFile` would be clearer ?

Reply:
Yeah, found it a tricky one to name. I also thought about `TimestampedFile` but I like your option more!